lib/src/floating_point_value.dart

```diff
@@ -0,0 +1,482 @@
+    if (other is! FloatingPointValue) {
+      return false;
+    }
+
+    if ((exponent.width != other.exponent.width) ||
+        (mantissa.width != other.mantissa.width)) {
+      return false;
+    }
+
+    return (sign == other.sign) &&
+        (exponent == other.exponent) &&
+        (mantissa == other.mantissa);
+  }
+
+  /// Return the value of the floating point number in a Dart [double] type.
+  double toDouble() {
+    var doubleVal = double.nan;
+    if (value.isValid) {
+      if (exponent.toInt() == 0) {
+        doubleVal = (sign.toBool() ? -1.0 : 1.0) *
+            pow(2.0, _eMin(exponent.width)) *
+            mantissa.toBigInt().toDouble() /
+            pow(2.0, mantissa.width);
+      } else if (exponent.toInt() !=
+          _eMax(exponent.width) + _bias(exponent.width) + 1) {
+        doubleVal = (sign.toBool() ? -1.0 : 1.0) *
+            (1.0 + mantissa.toBigInt().toDouble() / pow(2.0, mantissa.width)) *
+            pow(2.0, exponent.toInt() - _bias(exponent.width));
+      }
+    }
+    return doubleVal;
+  }
+
+  @override
+  String toString() => '${sign.toString(includeWidth: false)}'
+      ' ${exponent.toString(includeWidth: false)}'
+      ' ${mantissa.toString(includeWidth: false)}';
+
+  FloatingPointValue _performOp(
+      FloatingPointValue other, double Function(double a, double b) op) {
+    // make sure multiplicand has the same sizes as this
+    if (mantissa.width != other.mantissa.width ||
+        exponent.width != other.exponent.width) {
+      throw RohdHclException('FloatingPointValue: '
+          'multiplicand must have the same mantissa and exponent widths');
+    }
+
+    return FloatingPointValue.fromDouble(op(toDouble(), other.toDouble()),
+        mantissaWidth: mantissa.width, exponentWidth: exponent.width);
+  }
+
+  /// Multiply operation for [FloatingPointValue]
+  FloatingPointValue operator *(FloatingPointValue multiplicand) =>
+      _performOp(multiplicand, (a, b) => a * b);
+
+  /// Addition operation for [FloatingPointValue]
+  FloatingPointValue operator +(FloatingPointValue addend) =>
+      _performOp(addend, (a, b) => a + b);
+
+  /// Divide operation for [FloatingPointValue]
+  FloatingPointValue operator /(FloatingPointValue divisor) =>
+      _performOp(divisor, (a, b) => a / b);
+
+  /// Subtract operation for [FloatingPointValue]
+  FloatingPointValue operator -(FloatingPointValue subend) =>
+      _performOp(subend, (a, b) => a - b);
+
+  /// Negate operation for [FloatingPointValue]
+  FloatingPointValue negate() => FloatingPointValue(
+      sign: sign.isZero ? LogicValue.one : LogicValue.zero,
+      exponent: exponent,
+      mantissa: mantissa);
+}
+
+/// A representation of a single precision floating point value
+class FloatingPoint32Value extends FloatingPointValue {
+  static const int _exponentWidth = 8;
+  static const int _mantissaWidth = 23;
+
+  /// return the exponent width
+  static int get exponentWidth => _exponentWidth;
```

Comment:
you could just make the `_exponentWidth` and `_mantissaWidth` public directly instead of providing getters since they are `const` and can't be changed anyways.